src/Hilke.KineticConvolution/Helpers/Vector.cs

```diff
@@ -0,0 +1,25 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Hilke.KineticConvolution.Helpers
+{
+    public class Vector<TAlgebraicNumber>
```

Comment:
Really, `Vector<T>` should implement directly `IReadOnlyList<T>`, instead of exposing a `Coordinates` property. `Vector<T>` should then hold the reference to the coordinates in a private field, and `Dimension` could go in favor of `IReadOnlyList<T>.Count`.

Reply:
The static factory method could be replaced by a constructor a bit more smart, such as:
```cs
public Vector(IEnumerable<TAlgebraicNumber> coordinates)
{
    Coordinates = coordinates as IReadOnlyList<TAlgebraicNumber> ?? coordinates.ToList();
}
```